### ISSUE TITLE
Cannot use Fira Code Nerd Font Retina with VScode

### ISSUE THREAD

**Issue by trungklam** [ryanoasis/nerd-fonts#440]
#### 🎯 Subject of the issue

I installed various type of Fira Code Nerd Font Complete but want to specify to use Retina as the font for my VSCode. The below setting didn't work:
`"editor.fontFamily": "'Fira Code Retina Nerd Font Complete'",`
or
`"editor.fontFamily": "FiraCode Nerd Font-Retina'"`
or
`"editor.fontFamily": "FiraCode Nerd Font Retina"`

The default Fira Code will work with:
`"editor.fontFamily": "FiraCode-Retina'"`
To use the regular, I use the below setting and It works:
`"editor.fontFamily": "FiraCode Nerd Font",`
#### 🔧 Your Setup
- Fira Code Retina Nerd Font Complete
- VSCode 1.42
- macOS 10.15.3

**Comment by triangularcover** [comment]
try "FiraCodeNerdFontComplete-Regular" as a font name (at least vsc does not fail-back to the next font), 
I found it using:
```bash
$ strings ~/Library/Fonts/"Fira Code Retina Nerd Font Complete.otf" | grep Fira | less
```

**Comment by Happin3ss** [comment]
> try "FiraCodeNerdFontComplete-Retina" as a font name (cannot visually distinguish it from Regular, but at least vsc does not fail-back to the next font). I found it using:
> 
> ```shell
> $ strings ~/Library/Fonts/"Fira Code Retina Nerd Font Complete.otf" | grep Fira | less
> ```

Thank you for help. I tried but it didn't work for me. 
`"editor.fontFamily": "FiraCodeNerdFontComplete-Retina"`

Seems the only way to use retina is only install it without any other version and use the below setting:
`"editor.fontFamily": "FiraCode Nerd Font Mono",`

**Comment by aQuaYi** [comment]
```text
	"editor.fontFamily": "'JetBrains Mono','Fira Code Retina',FiraCode-Retina",
```

**Comment by wizardlink** [comment]
@Happin3ss, my issue was not with VSCode specifically but this may help you.

I were checking what name Fira Code had in my system and it was called `FiraCode NF`, from there I used `FiraCode NF Retina` and it worked properly! I am unsure whether this may be your issue or not, but I hope I have helped you or anyone else with this issue.

**Comment by SharakPL** [comment]
Just like @wizardlink mentioned, the name of this font is `FiraCode NF` and `FiraCode NF Retina` so to use it in VSCode your font setting should be like this:
> "editor.fontFamily": "FiraCode NF Retina"

**Comment by albanv** [comment]
for me, it worked with `FiraCodeNerdFontCompleteM-Retina` (`M` stands for the Mono version)

**Comment by saulimus** [comment]
I installed mine using homebrew just today:
```
  brew tap homebrew/cask-fonts
  brew cask install font-fira-code-nerd-font
```
I reached the same conclusion as @triangularcover but using a different method.
I checked the output of the command mentioned here: https://apple.stackexchange.com/a/243746
Then found the entry for "Fira Code Retina Nerd Font Complete.otf" and used the value "FiraCodeNerdFontComplete-Retina" found below "Typefaces". I'm not sure qhy for some the font is called `FiraCode NF`, maybe an older version?

Example of output:
```
      Kind: OpenType
      Valid: Yes
      Enabled: Yes
      Location: /Users/sauls/Library/Fonts/Fira Code Retina Nerd Font Complete.otf
      Typefaces:
        FiraCodeNerdFontComplete-Retina:
          Full Name: Fira Code Retina Nerd Font Complete
          Family: FiraCode Nerd Font
          Style: Retina
          Version: Version 2.000;PS 002.000;hotconv 1.0.88;makeotf.lib2.5.64775
          Vendor: Carrois Corporate, Edenspiekermann AG, Nikita Prokopov
          Unique Name: 2.000;CTDB;FiraCode-Retina
          Designer: Carrois Corporate, Edenspiekermann AG, Nikita Prokopov
          Copyright: Copyright 2012-2015 The Fira Code Project Authors (https://github.com/tonsky/FiraCode)
          Trademark: Fira Mono is a trademark of The Mozilla Corporation.
          Outline: Yes
          Valid: Yes
          Enabled: Yes
          Duplicate: No
          Copy Protected: No
          Embeddable: Yes
```

**Comment by AvyChanna** [comment]
Linux users can change the font family with `fontforge` as a workaround.

Goto `Element>Font Info`. In the new window, change `PS Names>Family Name` and `TTF Names>Preferred Family`
Then `File>Generate Font>Generate`. Select Font Type(I used TrueType). Ignore warnings, if any.

After placing them in the appropriate folder(like ~/.local/share/fonts/), rebuild font cache with `fc-cache -f ~/.local/share/fonts/`.

You can check if everything went right by `c-list | grep -i "fira" | column -t -s":" `(Displays - font  family  stype)
```plaintext
# Sample Output
/home/h2g2/.local/share/fonts/Fira Code Bold Nerd Font Complete Mono.ttf      FiraCode Nerd Font Mono         style=Bold
/home/h2g2/.local/share/fonts/Fira Code Retina Nerd Font Complete Mono.ttf    FiraCode Nerd Font Mono Retina  style=Retina,Regular
/home/h2g2/.local/share/fonts/Fira Code Medium Nerd Font Complete Mono.ttf    FiraCode Nerd Font Mono Medium  style=Medium,Regular
/home/h2g2/.local/share/fonts/Fira Code Regular Nerd Font Complete Mono.ttf   FiraCode Nerd Font Mono         style=Regular
/home/h2g2/.local/share/fonts/Fira Code Light Nerd Font Complete Mono.ttf     FiraCode Nerd Font Mono Light   style=Light,Regular
```
Now you can use `"editor.fontFamily": "FiraCode Nerd Font Mono Retina"` in vscode.

**Comment by vt4075** [comment]
On macOS:

Used `$ system_profiler SPFontsDataType >> fonts.txt` to generate a text file containing information on all the fonts in my computer.

Search for which `Family` name the font belongs to:

```
      Typefaces:
        FiraCodeNerdFontCompleteM-Light:
          Full Name: Fira Code Light Nerd Font Complete Mono
          Family: FiraCode Nerd Font Mono
          Style: Light
```

Added the following line to `settings.json`:
`"terminal.integrated.fontFamily": "FiraCode Nerd Font Mono"`

Keep in mind that VSCode terminal only accepts monocode fonts

**Comment by shelbyspeegle** [comment]
For what its worth - `"terminal.integrated.fontFamily": "HackNerdFontComplete-Regular",` ended up working for me.

**Comment by FHXISDOG** [comment]
> for me, it worked with `FiraCodeNerdFontCompleteM-Retina` (`M` stands for the Mono version)

Thanks!It' work for me !

**Comment by rakshit087** [comment]
> Download Nerd Font, but only install **FiraCode Nerd Font Retina Complete.** Then go to the font settings of your windows 10 and see what is the display name of the font. Then in the settings.json file `"editor.fontFamily": your font name with retina`
> For Me: `"editor.fontFamily": "FiraCode Nerd Font Retina"`

Worked for me :) Thank You :+1:

**Comment by mtsitzer** [comment]
I'm seeing an issue with FiraCode NF Retina not being recognized as a distinct font.

This suggests its part of the FiraCode NF Family
<img width="549" alt="image" src="https://user-images.githubusercontent.com/28517224/147736019-9ba59525-ec9d-4746-a7db-193062a9304d.png">

But when I go into the fonts folder, I don't see Retina as a distinct font, nor do I see it as part of the family
<img width="434" alt="image" src="https://user-images.githubusercontent.com/28517224/147735890-7fe310f3-a749-490d-be3e-b8ffdc331dd2.png">

Compare this with the standard (non Nerd Font) Fira Code:
<img width="394" alt="image" src="https://user-images.githubusercontent.com/28517224/147735955-a2698d1f-e61f-4f00-8f68-7cc156e9ea53.png">

**Comment by KavehKarami** [comment]
I've used 'FiraCode Nerd Font', it works for me in ubuntu

**Comment by KavehKarami** [comment]
I've used 'FiraCode Nerd Font', it works for me in ubuntu

**Comment by Finii** [comment]
Would be fixed by #717

**Comment by Finii** [comment]
@saulimus `FiraCode NF` is the `Windows Compatible` name.

Anyhow, the naming is broken for applications that use the old-style sets-of-4 with Family and Subfamily.
Newer applications that use TypographicFamily etc are fine already:

![image](https://user-images.githubusercontent.com/16012374/216776413-bf353f02-e622-4636-b127-ae71c46850f7.png)

You see hat the left half names (old system) are all equal :-(
This will be fixed with `v3.0.0` and `--makegroups`.

**Comment by haloxito1819** [comment]
There's a post with instructions to enable either Fira Code or Fira Code Retina in vscode:
https://github.com/tonsky/FiraCode/wiki/VS-Code-Instructions
